### PR TITLE
Implement CollabEditor

### PR DESCRIPTION
**Pull request by nonumpa**
## Text editor
  - Use [prosemirror](https://prosemirror.net/examples/basic/) as frontend text editor 
    - Use packaged schema: `prosemirror-schema-basic`, we can write our own schema when need [extra features](https://prosemirror.net/examples/schema/))
    - [placeholder plugin](https://discuss.prosemirror.net/t/how-to-input-like-placeholder-behavior/705)

## Backend 
  - [cofacts/collab-server](https://github.com/cofacts/collab-server): use [yjs](https://github.com/yjs/yjs), [hocuspocus](https://github.com/ueberdosis/hocuspocus) as collaboration backend which is compatible with prosemirror

#### Article list page
image

#### Article page
image

#### Article page
image
placeholder
image
online user
image

#### Article list page
image

#### Article page
image

### CODE REVIEW

File: components/Collaborate/CollabEditor.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        // console.log(node.type.name);
+        if (node.textContent) {
+          text += node.textContent;
+        }
+        text += '\n';
+      });
+
+      // TODO: listen textChanged event?
+      article.text = text;
+      editorView.destroy();
```
Comment (MrOrz) [inline components/Collaborate/CollabEditor.js:152]: Question: do we need to do anything to ensure that changes are all synced to Hocuspocus before destroying the prosemirror instance here?

File: components/Collaborate/Prosemirror.css
Comment (MrOrz) [inline components/Collaborate/Prosemirror.css:1]: Question: should we also include the CSS for awareness documented here?
https://github.com/yjs/y-prosemirror#remote-cursors

File: components/Collaborate/CollabEditor.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        // console.log(node.textContent);
+        // console.log(node.type.name);
+        if (node.textContent) {
+          text += node.textContent;
+        }
+        text += '\n';
+      });
+
+      // TODO: listen textChanged event?
+      article.text = text;
```
[the remark this answers is not in the document]
Comment (nonumpa) [reply components/Collaborate/CollabEditor.js:151]: Yes, but I am not sure if client can know when the article.text is updated. Maybe we can try [HocuspocusProvider Events](https://tiptap.dev/hocuspocus/provider/events).

**Review by MrOrz — APPROVED** [review]
Let's test it on staging tonight~

**Pull request merged**